File: library-db/src/main/java/com/my/library_db/dao/UserDao.java
package com.my.library_db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.my.library_db.model.User;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface UserDao {
    @Query("SELECT * FROM user")
    Flowable<List<User>> getAll();

    @Query("SELECT * FROM user WHERE uid IN (:userIds)")
    Flowable<List<User>> loadAllByIds(int[] userIds);

    @Query("SELECT * FROM user WHERE first_name LIKE :first AND " +
            "last_name LIKE :last LIMIT 1")
    Flowable<User> findByName(String first, String last);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(User... users);

    @Delete
    void delete(User user);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateUsers(User... users);
}

File: library-db/src/main/java/com/my/library_db/dao/UserDao.kt
package com.my.library_db.dao

import androidx.room.*
import com.my.library_db.model.User
import io.reactivex.Flowable

@Dao
interface UserDao {
    @get:Query("SELECT * FROM user")
    val all: Flowable<List<User?>?>

    @Query("SELECT * FROM user WHERE uid IN (:userIds)")
    fun loadAllByIds(userIds: IntArray?): Flowable<List<User?>?>?

    @Query("SELECT * FROM user WHERE first_name LIKE :first AND " +
            "last_name LIKE :last LIMIT 1")
    fun findByName(first: String?, last: String?): Flowable<User?>?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(vararg users: User?)

    @Delete
    fun delete(user: User?)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateUsers(vararg users: User?)
}